dojo/tools/gitlab_dast/parser.py

```diff
@@ -0,0 +1,163 @@
+import json
+import hashlib
+from datetime import datetime
+from dojo.models import Finding, Endpoint
+
+
+class GitlabDastParser(object):
+    """
+    Import GitLab DAST Report in JSON format
+    """
+
+    def get_scan_types(self):
+        return ["GitLab DAST Report"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return "GitLab DAST Report"
+
+    def get_description_for_scan_types(self, scan_type):
+        return "GitLab DAST Report in JSON format (option --json)."
+
+    # turning a json file to string
+    def parse_json(self, file):
+        data = file.read()
+        try:
+            tree = json.loads(str(data, "utf-8"))
+        except:
+            tree = json.loads(data)
+
+        return tree
+
+    def get_items(self, tree, test):
+        items = {}
+
+        # iterating through each vulnerability
+        for node in tree["vulnerabilities"]:
+            item = get_item(node, test)
+
+            item_key = hashlib.sha256(
+                "|".join([item.severity, item.title, item.description]).encode()
+            ).hexdigest()
+
+            if item_key in items:
+                items[item_key].unsaved_endpoints.extend(item.unsaved_endpoints)
+                items[item_key].nb_occurences += 1
+            else:
+                items[item_key] = item
+
+        return list(items.values())
+
+    def get_findings(self, file, test):
+        if file is None:
+            return None
+
+        tree = self.parse_json(file)
+        if tree:
+            return self.get_items(tree, test)
+
+    def convert_severity(self, num_severity):
+        """Convert severity value"""
+        if num_severity >= -10:
+            return "Low"
+        elif -11 >= num_severity > -26:
+            return "Medium"
+        elif num_severity <= -26:
+            return "High"
+        else:
+            return "Info"
+
+
+# iterating through properties of each vulnerability
+def get_item(vuln, test):
+
+    if vuln["category"] != "dast":
+        return None
+
+    # scanner_confidence
+    scanner_confidence = get_confidence_numeric(vuln["confidence"])
+
+    # id
+    if "id" in vuln:
+        unique_id_from_tool = vuln["id"]
+    else:  # deprecated
+        unique_id_from_tool = vuln["cve"]
```

Comment:
This is because the schema from GitLab DAST says that the id field is not mandatory.
Ref: https://gitlab.com/gitlab-org/security-products/security-report-schemas/-/blob/master/dist/dast-report-format.json

Although I do agree that the id field should never be empty, but I kinda want to conform to the schema provided by Gitlab.

Reply:
OK, I decided to remove it entirely because I do agree it kind of does not make sense to use cve as an id.